common:
    lid_topic:  "/velodyne_points"
    imu_topic:  "/imu"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible
    time_offset_lidar_to_imu: 0.0 # Time offset between lidar and IMU calibrated by other algorithms, e.g. LI-Init (can be found in README).
                                  # This param will take effect no matter what time_sync_en is. So if the time offset is not known exactly, please set as 0.0

preprocess:
    lidar_type: 2                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 
    scan_line: 32
    scan_rate: 10                # only need to be set for velodyne, unit: Hz,
    timestamp_unit: 2            # the unit of time/t field in the PointCloud2 rostopic: 0-second, 1-milisecond, 2-microsecond, 3-nanosecond.
    blind: 2

mapping:
    acc_cov: 0.1
    gyr_cov: 0.1
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    # acc_cov: 1.1197412605492375e-02 
    # gyr_cov: 1.0270904839480961e-02 
    # b_acc_cov: 1.1751767903346351e-04
    # b_gyr_cov: 9.1355383994881894e-05
    fov_degree:    180
    det_range:     100.0
    extrinsic_est_en:  false      # true: enable the online estimation of IMU-LiDAR extrinsic,
    extrinsic_T: [ 0.24434680,  0.70243411,  0.43841361]
    extrinsic_R: [ 0.0724, -0.9974, 0.0006, 
                   0.9973, 0.0724, -0.0112, 
                   0.011, 0.0014, 0.9999]

publish:
    path_en:  false
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.

cov:
    lidar_point_cov: 0.05
    # 初始协方差阵，顺序：位置、姿态、外参矩阵、杆臂、速度、陀螺仪零偏、加速度计零偏、重力、虚拟零偏
    # init_P: [1, 1, 0.00001, 0.00001, 1, 0.0001, 0.001, 0.00001, 0.00001] 初始
    # init_P: [1, 1, 0.00001, 0.00001, 1, 0.00001, 0.0001, 0.00001, 0.00001] -
    # init_P: [1, 1, 0.00001, 0.00001, 1, 0.00001, 0.00001, 0.00001, 0.00001] --
    init_P: [1, 1, 0.00001, 0.00001, 1, 0.00001, 0.00001, 0.001, 0.00001]

others:
    grav_S2: true
    is_extended: false